src/Illuminate/Foundation/Support/Providers/RouteServiceProvider.php

```diff
@@ -29,6 +29,12 @@ public function boot(Router $router)
         } else {
             $this->loadRoutes();
         }
+
+        // We still want name look-ups to be fast, even if names were specified fluently
+        // on the route itself. This will update the name look-up table just in case.
+        $this->app->booted(function () use ($router) {
+            $router->getRoutes()->refreshNameLookup();
+        });
```

Comment:
Why not put it as separate method (or make it a trait), which you can call to enable this feature.